client/js/widgets/widget.js

```diff
@@ -1497,16 +1497,22 @@ export class Widget extends StateManaged {
           if(typeof nextTurnIndex != 'number' || !isFinite(nextTurnIndex))
             nextTurnIndex = 0;
           const turn = indexList[mod(nextTurnIndex, indexList.length)];
+          variables.TURN = turn;
```

Comment:
I don't think this works for `random` (and in fact in the pr-test room it doesn't). I *think* you are intending to have `variables.TURN` contain the new saved turn. But on this line, you have to set it if `readOnly` is true, and clear it if `readOnly` is false. Then, in the `a.turnCycle == 'random'` branch, you need to test `variables.TURN` and either just use it or call `random`.

Reply:
I still think the way it is at the moment is good enough as described in my previous coment

Reply:
Wait, I don't think you can leave it as is. It just doesn't work for `random` (that is, if you run `readOnly` with `random`, and then run it for real, you will likely get different results. I think I proposed a fix above that would work.